internal/session/repository_connection.go

```diff
@@ -166,31 +165,29 @@ func (r *ConnectionRepository) AuthorizeConnection(ctx context.Context, sessionI
 			if err := reader.LookupById(ctx, &connection); err != nil {
 				return errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("failed for session %s", sessionId)))
 			}
-			connectionStates, err = fetchConnectionStates(ctx, reader, connectionId, db.WithOrder("start_time desc"))
-			if err != nil {
-				return errors.Wrap(ctx, err, op)
-			}
+
 			return nil
 		},
 	)
 	if err != nil {
-		return nil, nil, errors.Wrap(ctx, err, op)
+		return nil, errors.Wrap(ctx, err, op)
 	}
+	connection.ConnectionStatus = StatusAuthorized.String()
```

Comment:
I don't think we should just set this status directly like this. We should retrieve it from the database in the same way we do in LookupConnection.

Also see my other comment about using a view, which would allow the `LookupById` call above to retrieve the status.